projects/js-packages/publicize-components/src/social-store/resolvers.js

```diff
@@ -72,6 +73,12 @@ export function getPostShareStatus( _postId ) {
 	return async ( { dispatch, registry } ) => {
 		// Default to the current post ID if none is provided.
 		const postId = _postId || registry.select( editorStore ).getCurrentPostId();
+		const featureFlags = registry.select( socialStore ).featureFlags();
```

Comment:
When accessing the same store, you can destructure `select` in that callback like `dispatch` and `registry` and then directly use `select.featureFlags()`

```ts
return async ( { dispatch, registry, select } ) => {
	const featureFlags = select.featureFlags();

	if ( ! featureFlags.useShareStatus ) {
		return;
	}

	// Rest of the logic
```